src/components/Hero.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState } from 'react'
+import key from '../assets/key.svg'
+import cmdK from '../assets/cmdK.svg'
+import starIcon from '../assets/starIcon.svg'
+import issueIcon from '../assets/issueIcon.svg'
+import { fetchWithSearch } from "../lib/supabase";
+import useDebounce from '../hooks/useDebounce'
+import humanizeNumber from "../lib/humanizeNumber";
+
+type PostResult = {
+  full_name: string;
+  user_id: number;
+  description: string;
+  stars: number;
+  issues: string;
+  contributions: { url: string; contributor: string; last_merged_at: string }[];
+};
+
+const Hero = () => {
+  const [searchTerm, setSearchTerm] = useState<string>("")
+  const debouncedSearchTerm: string = useDebounce<string>(searchTerm, 500);
+  const [results, setResults] = useState<PostResult[]>([]);
+  const [hasFocus, setFocus] = useState<boolean>(false);
+  
+  
+  useEffect(() => {
+    if (debouncedSearchTerm) {
+
+      fetchWithSearch("stars", 3, debouncedSearchTerm).then((results) => {
+        console.log(results)
+        setResults(results as unknown as []);
+      });
+    } else {
+      setResults([]);
+    }
+  }, [debouncedSearchTerm])
+  
+
+  return (
+    <div className='flex flex-col my-[95px] items-center'>
+        <div>
+          <h1 className='font-Lexend text-[48px] text-center text-lightSlate leading-[52.8px] tracking-[-6%] '>
+            Find <span className='bg-gradient-to-r from-gradFirst via-gradMiddle to-gradLast bg-clip-text text-transparent' >Open-Source Repositories</span> <br/> to  contribute today
+          </h1>
+        </div>
+        <div className='mt-[45px] px-[15px] gap-x-[10px] py-[10px] justify-between bg-white shadow-[0_35px_60px_-15px_rgba(0,0,0,0.3)] rounded-[16px] min-w-[422px]  flex'>
+          <img src={key} alt="key" />
+          <input
+            onFocus={() => setFocus(true)}
+            onBlur={() =>
+              setTimeout(() => {
+                setFocus(false);
+              }, 200)
+            }
+            onChange={(e => setSearchTerm(e.target.value))}
+            type="text" placeholder='Search repositories'
+            className='w-full outline-none text-[16px] text-lightSlate'
+          />
+          <img className='pt-[7px]' src={cmdK} alt="command k" />
```

Comment:
It would be best to not add this until this [issue](https://github.com/open-sauced/hot/issues/205) works.

Alternatively, you can leave it in and address that issue in your PR.